pandas/indexes/multi.py

```diff
@@ -955,15 +951,13 @@ def from_product(cls, iterables, sortorder=None, names=None):
         MultiIndex.from_arrays : Convert list of arrays to MultiIndex
         MultiIndex.from_tuples : Convert list of tuples to MultiIndex
         """
-        from pandas.core.categorical import Categorical
         from pandas.tools.util import cartesian_product
 
-        categoricals = [Categorical.from_array(it, ordered=True)
-                        for it in iterables]
-        labels = cartesian_product([c.codes for c in categoricals])
+        labels, levels = zip(*[_factorize(it) for it in iterables])
```

Comment:
e.g. doing the same thing here, nicer not to have to ``zip(*[_factorize(it) for it in iterables])``, rather
``_factorize_from_iterables``  (that calls ``_factorize`` internally)